libstuff/SSignal.cpp

```diff
@@ -173,31 +182,57 @@ void _SSignal_StackTrace(int signum, siginfo_t *info, void *ucontext) {
         if (!_SSignal_threadCaughtSignalNumber) {
             _SSignal_threadCaughtSignalNumber = signum;
 
+            SWARN("Signal " << strsignal(_SSignal_threadCaughtSignalNumber) << "(" << _SSignal_threadCaughtSignalNumber << ") caused crash, logging stack trace.");
+
             // What we'd like to do here is log a stack trace to syslog. Unfortunately, neither computing the stack
             // trace nor logging to to syslog are signal safe, so we try a couple things, doing as little as possible,
             // and hope that they work (they usually do, though it's not guaranteed).
 
             // Build the callstack. Not signal-safe, so hopefully it works.
-            void* callstack[100];
-            int depth = backtrace(callstack, 100);
-
-            // Log it to a file. Everything in this block should be signal-safe, if we managed to generate the
-            // backtrace in the first place.
-            int fd = creat("/tmp/bedrock_crash.log", 0666);
-            if (fd != -1) {
-                backtrace_symbols_fd(callstack, depth, fd);
-                close(fd);
+            void** callstack{0};
+            int max_depth = 10;
+            int depth{0};
+            while (true) {
+                if (callstack) {
+                    free(callstack);
+                }
+                callstack = (void**)malloc(sizeof(void*) * max_depth);
+                depth = backtrace(callstack, max_depth);
+                if (depth == max_depth) {
+                    max_depth *= 2;
```

Comment:
can you explain what this does. And why is max_depth originally set as 10. Is that not low?
Asked differently why not 100 which is what we started with before right?

Reply:
Cool and we do it in the loop like this to be more memory efficient instead of `depth = backtrace(callstack, 1000);`?
Is there an upper limit to stop? like if max_depth is > 1000 or is there no reason for this upperlimit?

Reply:
The callstack in the example logs above is 261,954 frames deep. The callstack we saw in production that we used to diagnose our performance issues was ~130,000 frames deep.

The most important part of the callstack is in the bottom few frames, if we limit this, we lose that.

The example case is:
```
somewhere in std::regex
somewhere in std::regex
somewhere in std::regex
somewhere in std::regex
... repeat the above 130,000 times ...
ReportActionUtils::getLastMessageTextFromHTML
Report::structureReportsForOnyx
OpenApp::childPostProcess
AuthCommand::postProcess
AuthCommand::postProcessCommand
AuthCommand::postProcess
BedrockCore::postProcessCommand
```

If you don't get to the bottom of the call stack, you can't tell where your recursion problem started, so we need to get the entire thing.

But in basically every other case, the stack in between 5 and 25 frames deep, we don't want to allocate space for half a million frames if we're not going to use it.

Reply:
okay got it. thanks for explaining 👍